### ISSUE TITLE
[Bug][BYT][APL] modprobe sof_pci_dev failed when testing unload /reload audio module.

### ISSUE THREAD

**Issue by Jiangxinx** [thesofproject/linux#1120]
***Describe the bug***
modprobe: FATAL: Module sof_pci_dev not found in directory /lib/modules/5.3.0-rc3-daily-20190809 on byt and apl platform.

***To Reproduce***
1.sudo ./sof_bootone.sh

***Actual result***
```
ubuntu@cd-apl-up2-pcm512x-07:~/scripts/kmod_scripts$ sudo ./sof_remove.sh 
Skipping sof_pci_dev, not loaded
Skipping sof_acpi_dev, not loaded
Removing snd_sof_intel_byt
rmmod: ERROR: Module snd_sof_intel_byt is in use by: snd_sof_pci
Skipping snd_sof_intel_hsw, not loaded
Skipping snd_sof_intel_bdw, not loaded
Removing snd_sof_intel_hda_common
rmmod: ERROR: Module snd_sof_intel_hda_common is in use by: snd_sof_pci
Removing snd_sof_intel_hda
rmmod: ERROR: Module snd_sof_intel_hda is in use by: snd_sof_intel_hda_common
Removing snd_sof_xtensa_dsp
rmmod: ERROR: Module snd_sof_xtensa_dsp is in use by: snd_sof_pci
Removing snd_soc_acpi_intel_match
rmmod: ERROR: Module snd_soc_acpi_intel_match is in use by: snd_sof_pci snd_sof_intel_hda_common
Skipping snd_soc_sst_bytcr_rt5640, not loaded
Skipping snd_soc_sst_bytcr_rt5651, not loaded
Skipping snd_soc_sst_cht_bsw_rt5645, not loaded
Skipping snd_soc_sst_cht_bsw_rt5670, not loaded
Skipping snd_soc_sst_byt_cht_da7213, not loaded
Removing snd_soc_sst_bxt_pcm512x
Skipping snd_soc_sst_bxt_tdf8532, not loaded
Skipping snd_soc_cnl_rt274, not loaded
Removing snd_sof
rmmod: ERROR: Module snd_sof is in use by: snd_sof_pci snd_sof_intel_hda_common snd_sof_intel_byt snd_sof_intel_ipc
Skipping snd_sof_nocodec, not loaded
Skipping snd_soc_wm8804_i2c, not loaded
Skipping snd_soc_wm8804, not loaded
Removing snd_soc_dmic
Skipping snd_soc_rt5670, not loaded
Skipping snd_soc_rt5645, not loaded
Skipping snd_soc_rt5651, not loaded
Skipping snd_soc_rt5640, not loaded
Skipping snd_soc_rl6231, not loaded
Skipping snd_soc_da7213, not loaded
Removing snd_soc_pcm512x_i2c
Removing snd_soc_pcm512x
Skipping snd_soc_tdf8532, not loaded
Skipping snd_soc_rt274, not loaded
Removing snd_soc_acpi
rmmod: ERROR: Module snd_soc_acpi is in use by: snd_sof_pci snd_soc_acpi_intel_match
Skipping snd_soc_icl_dummy, not loaded
Skipping snd_soc_skl_hda_dsp, not loaded
Removing snd_soc_hdac_hdmi
Skipping snd_soc_hdac_hda, not loaded
Removing snd_hda_ext_core
rmmod: ERROR: Module snd_hda_ext_core is in use by: snd_sof_intel_hda_common snd_sof_intel_hda
Skipping snd_hda_codec, not loaded
Removing snd_hda_core
rmmod: ERROR: Module snd_hda_core is in use by: snd_hda_ext_core snd_sof_intel_hda_common snd_sof_intel_hda
Removing snd_soc_core
rmmod: ERROR: Module snd_soc_core is in use by: snd_sof snd_sof_intel_hda_common
Removing snd_seq_midi
Removing snd_seq_midi_event
Removing snd_rawmidi
Skipping snd_pcm_dmaengine, not loaded
Removing snd_pcm
rmmod: ERROR: Module snd_pcm is in use by: snd_hda_ext_core snd_sof snd_sof_intel_hda_common snd_soc_core snd_hda_core
ubuntu@cd-apl-up2-pcm512x-07:~/scripts/kmod_scripts$ aplay -l
aplay: device_list:270: no soundcards found...
ubuntu@cd-apl-up2-pcm512x-07:~/scripts/kmod_scripts$ sudo ./sof_insert.sh apl snd_soc_pcm512x_i2c
modprobing snd_soc_pcm512x_i2c
modprobing sof_pci_dev
modprobe: FATAL: Module sof_pci_dev not found in directory /lib/modules/5.3.0-rc3-daily-20190809
```

***dmesg***
```
[  136.184773] sof-audio-pci 0000:00:0e.0: booting DSP firmware
[  136.184849] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x40000 successful
[  136.184865] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x40000 successful
[  136.184870] sof-audio-pci 0000:00:0e.0: period_bytes:0x0
[  136.184872] sof-audio-pci 0000:00:0e.0: periods:1
[  136.184937] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x3030303 successful
[  136.184947] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x3030302 successful
[  136.184951] sof-audio-pci 0000:00:0e.0: unstall/run core: core_mask = 1
[  136.184956] sof-audio-pci 0000:00:0e.0: DSP core(s) enabled? 1 : core_mask 1
[  136.185509] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x40000000 successful
[  136.185516] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x1010202 successful
[  136.201806] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x5000001 successful
[  136.203316] ------------[ cut here ]------------
[  136.203354] WARNING: CPU: 2 PID: 60 at /home/test/jenkins/workspace/kernel_build_template/sound/core/pcm_lib.c:1785 snd_pcm_period_elapsed+0xa3/0xb0 [snd_pcm]
[  136.203355] Modules linked in: snd_soc_sst_bxt_pcm512x snd_soc_hdac_hdmi snd_soc_dmic snd_sof_pci snd_sof_intel_hda_common snd_sof_intel_hda snd_sof_intel_byt snd_soc_acpi_intel_match snd_sof_intel_ipc snd_sof snd_sof_xtensa_dsp snd_soc_pcm512x_i2c snd_soc_acpi snd_soc_pcm512x snd_hda_ext_core snd_hda_core snd_soc_core snd_pcm snd_seq_midi snd_seq_midi_event snd_rawmidi x86_pkg_temp_thermal snd_seq snd_seq_device snd_timer mei_me snd intel_lpss_pci intel_lpss mei mfd_core soundcore efivarfs mmc_block i915 sdhci_pci cqhci i2c_algo_bit sdhci drm_kms_helper syscopyarea xhci_pci sysfillrect sysimgblt fb_sys_fops xhci_hcd drm
[  136.203382] CPU: 2 PID: 60 Comm: kworker/2:1 Tainted: G        W         5.3.0-rc3-daily-20190809 #51e23aea
[  136.203383] Hardware name: AAEON UP-APL01/UP-APL01, BIOS UPA1AM40 08/06/2018
[  136.203388] Workqueue: events sof_pcm_period_elapsed_work [snd_sof]
[  136.203393] RIP: 0010:snd_pcm_period_elapsed+0xa3/0xb0 [snd_pcm]
[  136.203396] Code: 78 ca f6 83 20 01 00 00 01 74 c1 48 8b bb 18 01 00 00 be 01 00 00 00 e8 bb 16 04 00 eb ae 48 c7 c7 d8 3c 79 c0 e8 46 b1 d2 de <0f> 0b 5b 5d 41 5c c3 66 0f 1f 44 00 00 83 fe 02 41 54 55 48 89 d5
[  136.203397] RSP: 0018:ffffa2630022fe80 EFLAGS: 00010282
[  136.203398] RAX: 0000000000000024 RBX: ffff9c2f33277a00 RCX: 0000000000000000
[  136.203399] RDX: 0000000000000000 RSI: ffff9c2f37b163d8 RDI: ffff9c2f37b163d8
[  136.203400] RBP: 0000000000000000 R08: 00000000000009bc R09: 0000000000000004
[  136.203401] R10: ffffa26300527dc8 R11: 0000000000000001 R12: ffff9c2f37b2c100
[  136.203402] R13: 0000000000000000 R14: ffff9c2f36132000 R15: 0ffff9c2f37b2c10
[  136.203403] FS:  0000000000000000(0000) GS:ffff9c2f37b00000(0000) knlGS:0000000000000000
[  136.203404] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  136.203405] CR2: 00007f58773ab290 CR3: 000000027579e000 CR4: 00000000003406e0
[  136.203406] Call Trace:
[  136.203415]  process_one_work+0x1e3/0x3d0
[  136.203418]  worker_thread+0x28/0x3c0
[  136.203420]  ? cancel_delayed_work+0x80/0x80
[  136.203422]  kthread+0x10e/0x130
[  136.203424]  ? kthread_park+0xa0/0xa0
[  136.203427]  ret_from_fork+0x35/0x40
[  136.203429] ---[ end trace c47455b9f65ebe7f ]---
[  136.215796] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x5 successful
[  136.215804] sof-audio-pci 0000:00:0e.0: FW Poll Status: reg=0x140000 successful
[  136.215807] sof-audio-pci 0000:00:0e.0: Firmware download successful, booting...
[  136.219143] sof-audio-pci 0000:00:0e.0: ipc rx: 0x70000000: FW_READY
```

***Environment***
Firmware: 62b59ce7 https://github.com/thesofproject/sof/commits/master
Kernel: 51e23aea https://github.com/thesofproject/linux/commits/topic/sof-dev
Topology: sof-apl-pcm512x.tplg & sof-byt-rt5651.tplg

***error log***
[dmesg.log](https://github.com/thesofproject/linux/files/3485052/dmesg.log)

**Comment by keyonjie** [comment]
@Jiangxinx the module name has been changed, please refer to sof PR: https://github.com/thesofproject/sof/pull/1712

**Comment by Jiangxinx** [comment]
Closed it with sof PR: thesofproject/sof#1712 .
